grommet/grommet
# Added support for custom drop overflow.

Currently Drop controls the overflow by default. In some specific circumstances it might be useful to turn off that overflow control and let things pop out. Consider this example:

![Screen Shot 2019-03-13 at 6 18 20 PM](https://user-images.githubusercontent.com/1207250/54324797-67411f00-45bc-11e9-953b-bcd745286b7d.png)

By default, this calendar would overflow under the drop contents, but with this PR the overflow behavior is turned off and the calendar pops out.

#### What does this PR do?

Allows users to override the default overflow auto in Drop.
This PR also adds support for `dropProps` for all components that uses `dropAlign`.
  * Menu
  * Select
  * TextInput
  * DropButton

#### Where should the reviewer start?

DropContainer has the bulk of the change.

#### What testing has been done on this PR?

manual

#### How should this be manually tested?
open menu, select, textinput, dropbutton and make sure everything is backwards compatible.
also, take a look at the overflow story for drop that allows the calendar to overflow (the whole reason for this PR).
#### Any background context you want to provide?

#### What are the relevant issues?

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
yes
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards. Although we are softly deprecating `dropAlign` and all other drop props from Menu, Select, TextInput, and DropButton in favor of `dropProps={}`.

So instead of doing:

```
<DropButton dropAlign={{ top: 'bottom' }} />
```

you do:

```
<DropButton dropProps={{ align: { top: 'bottom' } }} />
```
